Question 1

In [1]:
import pandas as pd

In [3]:
sneaker = pd.read_csv("2019 Winter Data Science Intern Challenge Data Set.csv")

In [4]:
sneaker

order_id  shop_id  user_id  order_amount  total_items payment_method  \
0            1       53      746           224            2           cash   
1            2       92      925            90            1           cash   
2            3       44      861           144            1           cash   
3            4       18      935           156            1    credit_card   
4            5       18      883           156            1    credit_card   
...        ...      ...      ...           ...          ...            ...   
4995      4996       73      993           330            2          debit   
4996      4997       48      789           234            2           cash   
4997      4998       56      867           351            3           cash   
4998      4999       60      825           354            2    credit_card   
4999      5000       44      734           288            2          debit   

               created_at  
0     2017-03-13 12:36:56  
1     2017-03-03 17:38:52  
2      2017-03-14 4:23:56  
3     2017-03-26 12:43:37  
4      2017-03-01 4:35:11  
...                   ...  
4995  2017-03-30 13:47:17  
4996  2017-03-16 20:36:16  
4997   2017-03-19 5:42:42  
4998  2017-03-16 14:51:18  
4999  2017-03-18 15:48:18  

[5000 rows x 7 columns]

In [7]:
order_amount = sneaker["order_amount"]

In [10]:
sum(order_amount)/5000

3145.128

a) Seems like the average order value (AOV) of $3145.13 was achieved because the order amount was not weighted properly to the number of items purchased.
It would not be fair comparing the price $351 (3 sneakers purchased) against $144 (1 sneakers purchased). This is because the first order purchased more sneakers than the second order.

In [11]:
sneaker["average_price"] = sneaker["order_amount"]/sneaker["total_items"]

b) The metric I would use instead would be to average the order amount by the total items purchased, creating a new column.

In [12]:
sneaker

order_id  shop_id  user_id  order_amount  total_items payment_method  \
0            1       53      746           224            2           cash   
1            2       92      925            90            1           cash   
2            3       44      861           144            1           cash   
3            4       18      935           156            1    credit_card   
4            5       18      883           156            1    credit_card   
...        ...      ...      ...           ...          ...            ...   
4995      4996       73      993           330            2          debit   
4996      4997       48      789           234            2           cash   
4997      4998       56      867           351            3           cash   
4998      4999       60      825           354            2    credit_card   
4999      5000       44      734           288            2          debit   

               created_at  average_price  
0     2017-03-13 12:36:56          112.0  
1     2017-03-03 17:38:52           90.0  
2      2017-03-14 4:23:56          144.0  
3     2017-03-26 12:43:37          156.0  
4      2017-03-01 4:35:11          156.0  
...                   ...            ...  
4995  2017-03-30 13:47:17          165.0  
4996  2017-03-16 20:36:16          117.0  
4997   2017-03-19 5:42:42          117.0  
4998  2017-03-16 14:51:18          177.0  
4999  2017-03-18 15:48:18          144.0  

[5000 rows x 8 columns]

In [13]:
sum(sneaker["average_price"])/5000

387.7428

c) The value obtained was $387.7 as the AOV

Question 2

In [ ]:
# a) How many orders were shipped by Speedy Express in total?

#SELECT COUNT(ShipperName) FROM
#(SELECT * FROM Shippers inner join Orders ON Shippers.ShipperID = Orders.ShipperID)
#WHERE ShipperName = 'Speedy Express';

#54 orders were shipped by Speedy Express in total.

In [ ]:
# b) What is the last name of the employee with the most orders?

#SELECT COUNT(*), LastName FROM
#(SELECT * FROM Employees inner join Orders ON Employees.EmployeeID = Orders.EmployeeID)
#Group BY LastName ORDER BY COUNT(*) desc;

#Employee with last name Peacock had the most orders which was 40 orders.

In [ ]:
# c) What product was ordered the most by customers in Germany?

#SELECT Count(*), ProductName, Country FROM 
#(SELECT * FROM Customers inner join Orders ON Customers.CustomerID = Orders.CustomerID) table1
#inner join (SELECT * FROM OrderDetails inner join Products ON OrderDetails.ProductID = Products.ProductID) table2
#ON table1.OrderID = table2.OrderID WHERE Country = 'Germany' GROUP BY ProductName ORDER BY Count(*) desc;

#The product ordered most by customers in Germany was Gorgonzola Telino.